updatesnap/SnapModule/snapmodule.py

```diff
@@ -115,6 +115,13 @@ def _get_version(self, part_name, entry, entry_format, check):
                                   "Missing tag version format for "
                                   f"{part_name}.")
             return None  # unknown format
+        # Check for variations in the version format and beta releases.
+        if '%V' in entry_format['format']:
+            version = pkg_resources.parse_version(entry.split("/")[-1])
```

Comment:
I don't see how this manages to only parse those entries that begin with the specified format. For example, if we specify that format is "debian%V", this code will return any entry, not only those that begin with "debian".